cranelift-codegen/meta/src/isa/x86/recipes.rs

```diff
@@ -481,6 +481,21 @@ pub fn define<'shared>(
             ),
     );
 
+    {
+        recipes.add_template_recipe(
+            EncodingRecipeBuilder::new("splat", f_unary, 2)
+                .operands_in(vec![fpr])
+                .operands_out(vec![fpr])
+                .emit(
+                    r#"
+                    {{PUT_OP}}(bits, rex2(in_reg0, out_reg0), sink);
+                    modrm_rr(in_reg0, out_reg0, sink);
+                    sink.put1(0);
```

Comment:
I'll rename this recipe to something else... but what? And the `sink.put1(0)` is a hardcoding that would be nice to avoid (see comments on `pshuf` above)

Reply:
If this can be reused for all the pshufw/d etc., maybe the name could be `pshuf_base` or something like this? (Or if you understand the preexistent naming scheme with `rr` etc, apply the scheme here too? I'm not too sure how this naming scheme works...)